resources/net.go

```diff
@@ -0,0 +1,804 @@
+			}
+			if obj.debug {
+				log.Printf("%s: Event(%s): %v", obj, event.Body.Name, event.Body.Op)
+			}
+
+			send = true
+			obj.StateOK(false) // dirty
+
+		case event, ok := <-svcChan:
+			if !ok {
+				closeMap[2] = true // done
+				continue
+			}
+			if obj.debug {
+				log.Printf("%s: Event(%s): %v", obj, event[networkdService].ActiveState, event)
+			}
+
+			send = true
+			obj.StateOK(false) // dirty
+
+		case err, ok := <-svcErr:
+			if !ok {
+				closeMap[3] = true // done
+				continue
+			}
+			return errwrap.Wrapf(err, "unknown %s error", obj)
+
+		case event, ok := <-obj.Events():
+			if !ok {
+				closeMap[4] = true // done
+				continue
+			}
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				return *exit // exit
+			}
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			send = false
+			obj.Event()
+		}
+	}
+}
+
+// ifaceCheckApply checks the state of the network device and performs the
+// appropriate actions to converge its state with the graph.
+func (obj *NetRes) ifaceCheckApply(apply bool) (checkOK bool, _ error) {
+	var ok bool
+
+	// create a bus object pointing to the adapter
+	busObj := obj.gdbus.Object(dBusNetworkdInterface, obj.ifacePath)
+	// get the object's operational state
+	call := busObj.Call(dBusPropertiesGet, 0, dBusNetworkdLinkInterface, opState)
+	if call.Err != nil {
+		return false, call.Err
+	}
+	// check the operational state
+	os, ok := call.Body[0].(dbus.Variant)
+	if !ok {
+		return false, fmt.Errorf("error casting opstate")
+	}
+
+	// if the interface's state is correct, we're done
+	if obj.State == ifaceDown && os == dbus.MakeVariant(opStateOff) {
+		return true, nil
+	}
+	if obj.State == ifaceUp {
+		if os != dbus.MakeVariant(opStateOff) {
+			return true, nil
+		}
+		// XXX: retry logic for fault tolerance
+		if os == dbus.MakeVariant(opStateNoCarrier) {
+			return false, fmt.Errorf("connection failed")
+		}
+	}
+
+	// if the service is restarting, wait for the next event
+	restart, err := obj.svcRestarting()
+	if err != nil {
+		return false, err
+	}
+	if restart {
+		return false, nil
+	}
+
+	if !apply {
+		return false, nil
+	}
+
+	log.Printf("%s: ifaceCheckApply(%t)", obj, apply)
+
+	// enable or disable the interface
+	cmd := exec.Command("ip", "link", "set", obj.GetName(), obj.State)
+
+	// open a pipe to get error messages from os/exec
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return false, errwrap.Wrapf(err, "failed to initialize stderr pipe")
+	}
+
+	// start the command
+	if err := cmd.Start(); err != nil {
+		return false, errwrap.Wrapf(err, "cmd failed to start")
+	}
+	// capture any error messages
+	slurp, err := ioutil.ReadAll(stderr)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error slurping error message")
+	}
+	// wait until cmd exits and return error message if any
+	if err := cmd.Wait(); err != nil {
+		return false, errwrap.Wrapf(err, "%s", slurp)
+	}
+	// XXX: wait for state?
```

Comment:
did you find out about the daemon restart stuff?

Reply:
Yes. It may "reload" the unit file, but the ip doesn't change until the service is restarted. tried `systemctl daemon-reload` and dbus.Conn.Reload() but the ip does not change without restarting the service.